### DISCUSSION TITLE
Custom hihat control pedal (TCRT5000)

### DISCUSSION THREAD

**Discussion by bzass** [corrados/edrumulus#39] [General]
Hi,

Finally I experimented with edrumulus today and I have to say that @corrados - HUGE kudos to you! I was very positively surprised with results. Latency is amazing. Very responsive and not overly chatty over MIDI CC - great job!

I haven't tried positional sensing yet - started from hihat connection (CY5 + custom hihat pedal based on TCRT5000). I used FD8 settings and it seems to be almost accurate (I'm getting MIDI notes from 0 to 115). I'm assuming that it should be 0-127 for everything to work properly. I tested readings from pedal input - they are between 67 and 913. Is there any way to map it to parameters to calibrate the pedal so it covers full spectrum of values (0-127)? Which parameters control it?

I'm assuming that this full spectrum is needed for closed / splash sound to work. Even on current settings I'm getting note 44 after very dynamic press but with very low velocity (eg. 8).

Thanks!

**Comment by bzass** [comment]
Just found out that sensitivity parameter allows increasing the upper range to make it 127 :)

**Comment by corrados** [comment]
> I'm assuming that this full spectrum is needed for closed / splash sound to work. Even on current settings I'm getting note 44 after very dynamic press but with very low velocity (eg. 8).
> [...]
> Just found out that sensitivity parameter allows increasing the upper range to make it 127 :)

Yes, the Sensitivity parameter is the correct one. Did this change solved the splash sound issue, too?
You should also play a bit with the Threshold parameter. This should define the lower end of the MIDI CC scale.

**Reply by bzass** [reply]
Partially yes, thank you - now there is a note after dynamic press. Not a splash but a closed sound but I will need to look into this more. Should splash be also sent as a note or is this emulated by midi receiver software?

**Reply by corrados** [reply]
Now I understand what you mean by "hi-hat splash sound". This is not supported in Edrumulus. Just the "closed sound".

**Reply by corrados** [reply]
Today, I did some tests with the splash sound. I did a test implementation on the following branch: https://github.com/corrados/edrumulus/compare/hi_hat_splash_sound_midi_note?expand=1

The idea was to output a different MIDI note if a hi-hat splash was detected by Edrumulus. To detect a splash pedal movement, it takes a quite large buffer for hi-hat pedal values to get it working. Also, it did not work very reliably.

But then I connected the current unmodified Edrumulus version to my Roland TD-27 and the splash sound worked out of the box without any modifications. So, I assume that the Roland module detects that the pedal was opened right after the pedal down sound and played an open hi-hat automatically.

So, it seems, the sound synthesizer (either a E-drum module or a software synthesizer) is in charge of supporting this feature and not the trigger module (i.e. Edrumulus).

**Reply by bzass** [reply]
Sorry for delay. Yes, I noticed the same (that it's being detected in the module/software instead of sending a midi note). I think that the closed note is completely good enough and it seems that other midi software can also emulate the splash sound on the software side - it seems to be the standard. I also think that this is just nice to have and not that critical. 🙂 thanks!
